pkg/querier/base/chunk_store_queryable.go

```diff
@@ -1,20 +1,22 @@
-package querier
+package base
```

Comment:
Wondering what's the actual reason for creating `base` package (instead of putting into `querier` package itself). too much names conflicts?

Correct me if I'm wrong. From what I see this base package is getting used only on two places `/querier/testutils.go` and `pkg/ruler/base/ruler_test.go`

Should we just keep in it `pkg/querier` instead? (not asking to do immediately, just wondering if I'm miss anything :))

Reply:
hmm good point. I double checked after your comment and I think the main issue is that `ruler_test.go` relies on `base.Querier`, and `base.Querier` itself requires all the other stuff.

Anyway, this is worth a try - I'll see how we can get rid of it. Maybe I can modify `ruler_test.go` to instead rely on `pkg/querier.Querier` instead of `pkg/querier/base.Querier`.

Reply:
Thanks @DylanGuedes for trying this :+1:

Reply:
Ok so after some time digging into it:
The reason why the whole `base` seems necessary is because of a `ruler/base` test. I tried working around that test and modifying it a little bit to only relying on `pkg/querier` but I didn't have much success especially because everything related to `Queryable/QueryableFunc` is only available at `pkg/querier/base` and moving them to `pkg/querier` required moving al the rest.

All that said, a way I can see we getting rid of `querier/base` is by rewriting the tests under `ruler/base` substantially so that they won't require it (`querier/base`), but that will require not relying on anything related to `Queryable` and similar stuff anymore. I'll see what can be done regarding that today and I'll update this thread. Let's see :crossed_fingers: